### PR TITLE
#1291 Add an 'filter syntax index' table to documentation

### LINKED ISSUE #1291: Add an 'filter syntax index' table to documentation
To be added to the bottom of docs/filters.md

A partial list can be found in #1151 

Columns: `keyword` `aliases`
keyword column should be sorted alphabetically, and each keyword can be hyperlinked to the explanation.

### PR DESCRIPTION
**Pull request by zhongwei-z**
This closes #1291. I added a third column for the aliases the parameter of some qualifiers.

### CODE REVIEW

**Comment by dariusf** [comment]
Looking good. I don't think we need to repeat the keyword in the second column though. Also, the third column is a great idea, but could we make the format a bit more consistent somehow, to differentiate the parameters from the aliases?

**Comment by zhongwei-z** [comment]
Ok, I've put the aliases in brackets after the parameters.

File: docs/filters.md [hunk 1]
```diff
@@ -321,3 +321,27 @@ HubTurbo's filter system is incompatible with GitHub's in a number of ways.
     + `closed:[date range]`
     + `mentions:[string]`
     + `commenter:[string]`
+
+## Cheatsheet
```
Comment (damithc) [inline docs/filters.md]: Add this to table of contents?

File: docs/filters.md [hunk 2]
```diff
@@ -321,3 +321,27 @@ HubTurbo's filter system is incompatible with GitHub's in a number of ways.
...
+|Keywords|Aliases|Params aliases|
```
Comment (damithc) [inline docs/filters.md]: @dariusf is `Params aliases` is the best name for this column?

File: docs/filters.md [hunk 3]
```diff
@@ -321,3 +321,27 @@ HubTurbo's filter system is incompatible with GitHub's in a number of ways.
...
+|:------:|:-----:|--------------|
+|[assignee](#assignee)|`as`|
```
Comment (damithc) [inline docs/filters.md]: Can give the first column also in `code` font?

File: docs/filters.md (hunk 2 above)
Comment (dariusf) [inline docs/filters.md]: Maybe the first column could be `Qualifiers`, and the third could be `Keywords`. We overload 'keywords' quite a lot but that's the most general term for these.

**Comment by zhongwei-z** [comment]
Changed those :)

File: docs/filters.md [hunk 4]
```diff
@@ -321,3 +322,27 @@ HubTurbo's filter system is incompatible with GitHub's in a number of ways.
...
+|[`keyword`](#keyword)||
+|[`label`](#label)|`l`|
+|[`milestone`](#milestone)|`m`|`current` (`curr`)|
+|[`no`](#no)|`n`|
+|[`repo`](#repo)|`r`|
+|[`sort`](#sort)|`s`|`comments` (`cm`), `repo` (`r`), `updated` (`u`), `date` (`d`), `id`, `assignee` (`as`), `state` (`status`, `st`), `milestones` (`m`)|
+|[`state`](#state)|`status`, `st`|`open` (`o`), `closed` (`c`)|
+|[`title`](#title)|`t`|
+|[`type`](#type)|`ty`|`issue` (`i`), `pullrequest` (`pr`, `p`)|
+|[`updated`](#updated)|`u`|
```
Comment (damithc) [inline docs/filters.md]: I guess there is no way to align test to left?
Anyway, can you beautify the code (using spacing) to make the table look like a table, at least to the extent that is possible?
```
|Qualifiers                    |Aliases      |Keywords   |
|:----------------------------:|:-----------:|-----------|
|[`assignee`](#assignee)       |`as`         |-----------|
```

**Comment by zhongwei-z** [comment]
Aligned to the left, and the table is more readable in Markdown.

**Comment by dariusf** [comment]
Minor thing for next time: the issue number in the PR title goes at the back.

**Pull request merged**